terraform/modules/public-load-balancer/path_restriction_rule.tf

```diff
@@ -0,0 +1,53 @@
+locals {
+  # Needed due to an AWS quota restriction (limit of 5 conditions per rule)
+  # In future we may need to request a service quota increase of
+  # "Rules per Application Load Balancer" from AWS.
+  rules = setproduct(var.restricted_paths, var.unrestricted_cidrs)
+}
+
+resource "aws_lb_listener_rule" "allow_unrestricted_cidrs" {
```

Comment:
This is quite complex and uses `count`, so it'll likely do some quite unintuitive things when we make changes to the allowlists. It might also have complex failure modes if someone adds one too many entries. I'm wondering if it'd be the lesser of evils just to define the aws_lb_listener_rules where we call the module - i.e. let the client customise the rules in the few cases where we need. I guess we still need some way to not have a ton of duplicated stuff for the basic office-only allowlist in staging/test. Perhaps a separate module for generating the forwarding rules might be the way to go? (thinking SRP) Could then evolve it a bit more easily over time?